test/test_common/environment.h

```diff
@@ -93,6 +93,12 @@ class TestEnvironment {
    */
   static const std::string& nullDevicePath();
 
+  /**
+   * Obtain platform specific new line character(s)
+   * @return const std::string& platform specific new line character(s)
+   */
+  static const std::string& newLine();
```

Comment:
I wonder if providing something like `std::string convertNewLines(const std::string& s)` would be easier to use rather than having to format strings every place this is an issue. The implementation would replace all `\n` with `\r\n` for windows and just return s otherwise. Callers could write 

```
EXPECT_EQ(waitForAccessLog(access_log_name_, 1),
          TestEnvironment::convertNewLines("RFCF cache.response_from_cache_filter\n"));
```

From a brief look through the code it seems like there might be more locations that will need to be converted eventually, but perhaps not. What do you think?

Reply:
> I wonder if providing something like std::string convertNewLines(const std::string& s) would be easier to use rather than having to format strings every place this is an issue.

No strong opinion on the API. I draw my inspiration from C# `Environment.NewLine` [source](https://docs.microsoft.com/en-us/dotnet/api/system.environment.newline?view=netcore-3.1) this is why I created this function. Personally I find it more intuitive but maybe but maybe I am just biased because I am used to it. I leave it up to you.

> From a brief look through the code it seems like there might be more locations that will need to be converted eventually, but perhaps not. What do you think?

I opted to add the API and fix a single instance here so we can get the functionality in the code and then we would fix it forward as we make the whole test pass Windows. I can aggressively change more places but I feel that when the function goes in it could be used as needed to fix the tests. Again up to you if you prefer to batch and change everything now or change it as we encounter test failures.

Reply:
It's fine to continue as-is, but we should apply @lizan's suggestion to make it a absl::string_view and you can just return `{"\n",1}` or `{"\r\n", 2}`

Reply:
fixed in bf596518c032a3a88657ef885bb7c0840eb348d2